# SI618 Project Part 2: Analysis

# Project Title — *World of Board Games: What decide their popularity and ratings?*

## Team Members
- Keqing Lu (keqinglu)
- Xiyuan Wang (denniswx)
- Fangqing Yuan (fangqing)

# Table of Contents
* [Descriptive Statistics](#Descriptive_Statistics)
    * [Section 1.1](#section_1_1)
* [Chapter 2](#chapter2)
    * [Section 2.1](#section_2_1)
        * [Sub Section 2.1.1](#sub_section_2_1_1)
        * [Sub Section 2.1.2](#sub_section_2_1_2)
* [Chapter 3](#chapter3)
    * [Section 3.1](#section_3_1)
        * [Sub Section 3.1.1](#sub_section_3_1_1)
        * [Sub Section 3.1.2](#sub_section_3_1_2)
    * [Section 3.2](#section_3_2)
        * [Sub Section 3.2.1](#sub_section_3_2_1)

In [2]:
# import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
# import processed data from project 1
ready_data = pd.read_csv('board_games.csv')

In [4]:
ready_data.head() # check the first few rows of the data

num_x     id         name  year  rank  average  bayes_average  users_rated  \
0    105  30549     Pandemic  2008   106     7.59          7.487       108975   
1    189    822  Carcassonne  2000   190     7.42          7.309       108738   
2    428     13        Catan  1995   429     7.14          6.970       108024   
3     72  68448    7 Wonders  2010    73     7.74          7.634        89982   
4    103  36218     Dominion  2008   104     7.61          7.499        81561   

                          url  \
0   /boardgame/30549/pandemic   
1  /boardgame/822/carcassonne   
2         /boardgame/13/catan   
3  /boardgame/68448/7-wonders   
4   /boardgame/36218/dominion   

                                           thumbnail  ...  \
0  https://cf.geekdo-images.com/S3ybV1LAp-8SnHIXL...  ...   
1  https://cf.geekdo-images.com/okM0dq_bEXnbyQTOv...  ...   
2  https://cf.geekdo-images.com/W3Bsga_uLP9kO91gZ...  ...   
3  https://cf.geekdo-images.com/RvFVTEpnbb4NM7k0I...  ...   
4  https://cf.geekdo-images.com/j6iQpZ4XkemZP07HN...  ...   

             designer_list                                     publisher_list  \
0         ['Matt Leacock']  ['Z-Man Games', 'Albi', 'Asmodee', 'Asmodee It...   
1   ['Klaus-Jürgen Wrede']  ['Hans im Glück', '999 Games', 'Albi', 'Bard C...   
2         ['Klaus Teuber']  ['KOSMOS', '999 Games', 'Albi', 'Asmodee', 'As...   
3        ['Antoine Bauza']  ['Repos Production', 'ADC Blackfire Entertainm...   
4  ['Donald X. Vaccarino']  ['Rio Grande Games', '999 Games', 'Albi', 'Bar...   

  category_count  family_count  mechanic_count  expansion_count  \
0              1             9               7                7   
1              3            11               3              155   
2              2             5               9               87   
3              5             7               5               16   
4              2             3               5               46   

   implementation_count  designer_count  publisher_count  artist_count  
0                    11               1               33             5  
1                    17               1               35             4  
2                    29               1               48            11  
3                     3               1               14             4  
4                     2               1               23             9  

[5 rows x 40 columns]

## Descriptive Statistics <a id="Descriptive_Statistics"></a>

After the preliminary data manipulation, we have the following fields of data:

In [5]:
ready_data.columns

Index(['num_x', 'id', 'name', 'year', 'rank', 'average', 'bayes_average',
       'users_rated', 'url', 'thumbnail', 'num_y', 'primary', 'description',
       'yearpublished', 'minplayers', 'maxplayers', 'playingtime',
       'minplaytime', 'maxplaytime', 'minage', 'owned', 'trading', 'wanting',
       'wishing', 'category_list', 'mechanic_list', 'family_list',
       'expansion_list', 'implementation_list', 'artist_list', 'designer_list',
       'publisher_list', 'category_count', 'family_count', 'mechanic_count',
       'expansion_count', 'implementation_count', 'designer_count',
       'publisher_count', 'artist_count'],
      dtype='object')

For the purpose of our analysis, which investigates the factors making the games popular and/or highly rated, we are mainly interested in the following data: rating of the games, described by `average` (including `bayes_average`); popularity of the games, described by `owned`, `trading`, `wanting`, `wishing` and `users_rated`; the numerical factors of the games, including `yearpublished`, player number `minplayers` and `maxplayers`, `playingtime` including `minplaytime` and `maxplaytime`, `minage`; and the categorical factors including game category, game mechanics, game family etc., and their counts for each game.

### Rating of the games

In [6]:
ready_data.average.describe()

count    21631.000000
mean         6.417249
std          0.929345
min          1.040000
25%          5.830000
50%          6.450000
75%          7.040000
max          9.570000
Name: average, dtype: float64

In [7]:
ready_data.bayes_average.describe()

count    21631.000000
mean         5.683664
std          0.366096
min          0.000000
25%          5.510000
50%          5.546000
75%          5.678000
max          8.511000
Name: bayes_average, dtype: float64

Here we can see that the `bayes_average` is overall lower, and has a smaller standard deviation (i.e. more closely packed for the games) compared with `average`.

### Popularity of the games

#### Game ownership

In [8]:
ready_data.owned.describe()

count     21631.000000
mean       1487.924553
std        5395.077773
min           0.000000
25%         150.000000
50%         322.000000
75%         903.500000
max      168364.000000
Name: owned, dtype: float64

We can see that the median number of owners for a game is just around 300, while there are games owned by hundreds of thousands, which significantly raise the mean value.

#### Player desire and game trading

In [9]:
ready_data.trading.describe()

count    21631.000000
mean        43.585965
std        102.410851
min          0.000000
25%          5.000000
50%         13.000000
75%         38.000000
max       2508.000000
Name: trading, dtype: float64

In [10]:
ready_data.wanting.describe()

count    21631.000000
mean        42.030373
std        117.940355
min          0.000000
25%          3.000000
50%          9.000000
75%         29.000000
max       2011.000000
Name: wanting, dtype: float64

In [11]:
ready_data.wishing.describe()

count    21631.000000
mean       233.655587
std        800.657809
min          0.000000
25%         14.000000
50%         39.000000
75%        131.000000
max      19325.000000
Name: wishing, dtype: float64

We can see for most games, there are a number of players "wishing" to have the game. A smaller portion of them are "wanting" the game, which may mean a stronger intention level of buying. Some copies of the game are currently being traded.

#### Number of players rated

In [12]:
ready_data.users_rated.describe()

count     21631.000000
mean        874.548518
std        3695.946026
min          30.000000
25%          57.000000
50%         124.000000
75%         397.000000
max      108975.000000
Name: users_rated, dtype: float64

Overall, the `user_rated` number is lower than `owned`, which means not every player who owns a game has rated it. Meanwhile, there exist games that are do not have current owners, but still have rating records.

### Numerical factors of the games

#### Year published

In [13]:
ready_data.yearpublished.describe()

count    21631.000000
mean      1986.094910
std        210.042496
min      -3500.000000
25%       2001.000000
50%       2011.000000
75%       2017.000000
max       2023.000000
Name: yearpublished, dtype: float64

We can see that most games are published within the 21st century. Meanwhile, there are some very ancient games, which lowers the mean value.

In [14]:
pd.set_option('display.max_colwidth', None)
ready_data[ready_data.yearpublished == -3500].description

3622    Senet is an ancient Egyptian board game similar to backgammon.  Each player has 5 pawns which travel around a 3 by 10 board in an S shaped path.  Movement is dictated by the throw of 4 sticks (black on one side and white on the other) which can produce a 1,2,3,4, or 6, (5 is not possible) depending on how they land..  As in backgammon, the objective is to bear all of one's pawns off first.  A couple of interesting differences is that there are safe and trap spaces on the board.  Also, players may block each others' progress as in Parchisi.  Senet sets are usually constructed from various woods and are often quite beautiful.  Senet is one of the oldest boardgames but unfortunately it fell out of use for a few thousand years and no ancient rules set has been discovered.  Undoubtably another tragic loss resulting from the Library of Alexandria burning to the ground.  As a result scholars has made several educated guesses as to the rules.  The most popular versions are in the files

For example, the most ancient game is "Senet", which dates back to Ancient Egypt.

In [15]:
ready_data.yearpublished.value_counts().head(10)

yearpublished
2017    1325
2018    1321
2016    1291
2019    1258
2015    1147
2014    1007
2020     897
2013     862
2012     831
2011     741
Name: count, dtype: int64

The year with most games published is 2017, while the top 10 years constitute the entire 2010s.

#### Player number

In [16]:
ready_data.minplayers.describe()

count    21631.000000
mean         2.007027
std          0.688957
min          0.000000
25%          2.000000
50%          2.000000
75%          2.000000
max         10.000000
Name: minplayers, dtype: float64

In [17]:
ready_data.minplayers.value_counts().head(10)

minplayers
2     14834
1      3675
3      2462
4       503
5        59
0        44
6        24
8        16
7        11
10        2
Name: count, dtype: int64

In [18]:
ready_data.maxplayers.describe()

count    21631.000000
mean         5.709491
std         15.102385
min          0.000000
25%          4.000000
50%          4.000000
75%          6.000000
max        999.000000
Name: maxplayers, dtype: float64

In [19]:
ready_data.maxplayers.value_counts().head(10)

maxplayers
4     6810
2     4287
6     3969
5     2987
8     1218
10     413
1      369
7      324
3      279
12     248
Name: count, dtype: int64

In [20]:
players_range = ready_data.maxplayers - ready_data.minplayers
players_range.describe()

count    21631.000000
mean         3.702464
std         15.063117
min         -4.000000
25%          2.000000
50%          2.000000
75%          4.000000
max        997.000000
dtype: float64

We can see that most games have a minimum player number of 2, while the number of maximum players commonly range from 2 to 8, with 4 being the most. There are some peculiar values, including 0 for `minplayers` and `maxplayers`, which may indicate data missing; 999 for `maxplayers`, which may or maynot be an error but is interesting (as if the game designer doesn't really set a upper limit); entries with `maxplayers` smaller than `minplayers`, which probably indicates faulty data.

#### Playing time

In [21]:
ready_data.playingtime.describe()

count    21631.000000
mean        90.509177
std        534.826511
min          0.000000
25%         25.000000
50%         45.000000
75%         90.000000
max      60000.000000
Name: playingtime, dtype: float64

In [22]:
ready_data.playingtime.value_counts().head(10)

playingtime
30     3804
60     3169
45     2182
20     2134
120    1704
90     1687
15     1292
180     837
10      796
0       774
Name: count, dtype: int64

In [23]:
ready_data.minplaytime.describe()

count    21631.000000
mean        63.647774
std        447.213702
min          0.000000
25%         20.000000
50%         30.000000
75%         60.000000
max      60000.000000
Name: minplaytime, dtype: float64

In [24]:
ready_data.minplaytime.value_counts().head(10)

minplaytime
30     4417
60     2991
20     2654
45     2166
15     1893
120    1292
10     1262
90     1206
180     669
0       642
Name: count, dtype: int64

In [25]:
ready_data.maxplaytime.describe()

count    21631.000000
mean        90.509177
std        534.826511
min          0.000000
25%         25.000000
50%         45.000000
75%         90.000000
max      60000.000000
Name: maxplaytime, dtype: float64

In [26]:
playtime_range = ready_data.maxplaytime - ready_data.minplaytime
playtime_range.describe()

count    21631.000000
mean        26.861403
std        276.221914
min       -450.000000
25%          0.000000
50%          0.000000
75%         15.000000
max      21060.000000
dtype: float64

In [27]:
ready_data[ready_data.playingtime == 60000].description

9553    This is a wargame like no other.  While the map area (5 full size maps) has been repeated in other North Africa games, the game is detailed to a degree no other game has come close to.  If using the full rules you keep track of every individual plane and pilot in the three year campaign. Each counter on the board representing a ground unit is composed of many units which are kept track of on logs.  Supplies are kept track of and dispersed in a very detailed manner.&#10;&#10;From the rulebook we read how to run a game. &quot;CNA is a logistically-oriented game, and its play requires not only a lot of attention to logistics, but, if you will, a logistically sound methodology.&quot; It is suggested that you have 5 persons per side with the following duties.&#10;&#10;Commander-in-Chief: responsible for strategic decisions and to settle intra-team disputes.&#10;&#10;Logistics Commander: In charge of all supplies.  Accepts supply requisitions from the others and keeps all informed of

Firstly, we notice that the distribution of `playingtime` is identical to that of `maxplaytime`, so they are actually the same field of data. We find that the playing time of most games range within 10-180 minutes. The median of `minplaytime` is 30 minutes, and that of `maxplaytime` is 45 minutes. Over half of the games have the same listed `minplaytime` and `maxplaytime`. Similar with player numbers, there are some peculiar data, including a 0 for `maxplaytime` (probably missing data), very large `maxplaytime` (from the description it is indeed an extremely complicated game), and `maxplaytime` smaller than `minplaytime` (probably faulty data).

### Categorical factors of games

In [28]:
def string_to_list(string, default=None):
    """
    Converts a string representation of a list to a list.
    string: a string with a list representation, e.g. "[1, 2, 3]"
    default: the default value to return if the string is null
    """
    # if the string is null, return an empty list
    if pd.isnull(string):
        if default is not None:
            return [default]
        return []
    return eval(string)

In [29]:
ready_data.category_list = ready_data.category_list.apply(string_to_list)
ready_data.mechanic_list = ready_data.mechanic_list.apply(string_to_list)
ready_data.publisher_list = ready_data.publisher_list.apply(string_to_list, default='Unknown')
ready_data.designer_list = ready_data.designer_list.apply(string_to_list, default='(Uncredited)')

#### Game category

In [30]:
ready_data.category_list.explode().value_counts().describe()

count      84.000000
mean      674.190476
std       901.588698
min         1.000000
25%       195.500000
50%       356.000000
75%       783.250000
max      6402.000000
Name: count, dtype: float64

In [31]:
ready_data.category_list.explode().value_counts().head(10)

category_list
Card Game            6402
Wargame              3820
Fantasy              2681
Party Game           1968
Dice                 1847
Science Fiction      1666
Fighting             1658
Children's Game      1635
Abstract Strategy    1545
Economic             1503
Name: count, dtype: int64

In [32]:
ready_data.category_count.describe()

count    21631.000000
mean         2.618094
std          1.380869
min          0.000000
25%          2.000000
50%          2.000000
75%          3.000000
max         14.000000
Name: category_count, dtype: float64

We can find that there are 84 unique game categories, the most common category is 'Card Game', and on average a game is afiliated with 2.62 categories.

#### Game mechanics

In [33]:
ready_data.mechanic_list.explode().value_counts().describe()

count     182.00000
mean      345.60989
std       742.08696
min         1.00000
25%        25.25000
50%        52.00000
75%       301.00000
max      6112.00000
Name: count, dtype: float64

In [34]:
ready_data.mechanic_list.explode().value_counts().head(10)

mechanic_list
Dice Rolling              6112
Hand Management           4421
Set Collection            2936
Variable Player Powers    2719
Hexagon Grid              2371
Simulation                2099
Card Drafting             1869
Tile Placement            1805
Modular Board             1697
Grid Movement             1635
Name: count, dtype: int64

In [35]:
ready_data.mechanic_count.describe()

count    21631.000000
mean         2.907910
std          2.189651
min          0.000000
25%          1.000000
50%          2.000000
75%          4.000000
max         20.000000
Name: mechanic_count, dtype: float64

There are 182 unique game mechanics, the most common mechanics is 'Dice Rolling', and on average a game has 2.91 mechanics.

#### Game publishers

In [36]:
ready_data.publisher_list.explode().value_counts().describe()

count    7068.000000
mean        8.268251
std        32.347544
min         1.000000
25%         1.000000
50%         1.000000
75%         4.000000
max       597.000000
Name: count, dtype: float64

In [37]:
ready_data.publisher_list.explode().value_counts().head(10)

publisher_list
Hasbro                        597
(Self-Published)              594
Asmodee                       571
Ravensburger                  520
Parker Brothers               486
(Web published)               485
Pegasus Spiele                480
999 Games                     477
Korea Boardgames Co., Ltd.    474
KOSMOS                        429
Name: count, dtype: int64

In [38]:
ready_data.publisher_count.describe()

count    21631.000000
mean         2.701678
std          5.448527
min          1.000000
25%          1.000000
50%          1.000000
75%          3.000000
max        207.000000
Name: publisher_count, dtype: float64

There are 7068 unique publishers. The publisher with the most published games is 'Hasbro'. On average a publisher published 8.27 games [without excluding categories like '(Self-Published)' or '(Web-Published)'], but over half of publishers just have 1 game recorded. On average a game is published by 2.70 publishers.

#### Game designers 

In [39]:
ready_data.designer_list.explode().value_counts().describe()

count    9796.000000
mean        3.019600
std        21.663193
min         1.000000
25%         1.000000
50%         1.000000
75%         2.000000
max      2041.000000
Name: count, dtype: float64

In [40]:
ready_data.designer_list.explode().value_counts().head(10)

designer_list
(Uncredited)       2041
Reiner Knizia       329
Joseph Miranda      134
Wolfgang Kramer     133
Richard H. Berg     111
Bruno Cathala        95
Martin Wallace       92
Jim Dunnigan         92
James Ernest         91
Dean Essig           86
Name: count, dtype: int64

In [41]:
ready_data.designer_count.describe()

count    21631.000000
mean         1.367482
std          0.768842
min          1.000000
25%          1.000000
50%          1.000000
75%          2.000000
max         21.000000
Name: designer_count, dtype: float64

There are 9796 unique designers. The designer with the most designed games, excluding '(Uncredited)', is 'Reiner Knizia'. On average a designer designed 3.02 games [without excluding '(Uncredited)']. On average, a game has 1.37 designers.

In [42]:
pd.reset_option('display.max_colwidth')

## Inferential Statistics

对于不同年份，category，公司，family，ANOVA test一下平均分的不同；

可以只选几个最多的类别

对于不互斥的类别，是否test会有问题

把numerical features里面corr最高的几组3列出来

## Graphical Analysis

把第一次改一改

画一个5x5的pariplot

自由发挥。。。

## Comparative Analysis

histogram / scatterplot, hue="???"

??? 可以是 age， year，player_num

## Multivariate Analysis

画pivot table，cross tabulation

也是按照age,year,player_num

或者 card game, non-card game

## Reference